src/components/Menu/Menu.tsx

```diff
@@ -159,10 +159,14 @@ const MenuItems = (props: MenuItemsProps) => {
     ...props,
     ...popperAttributes,
   };
-
-  return (
-    <>{createPortal(<HeadlessMenu.Items {...optionProps} />, document.body)}</>
-  );
+  if (typeof window === 'object') {
+    return (
+      <>
+        {createPortal(<HeadlessMenu.Items {...optionProps} />, document.body)}
+      </>
+    );
+  }
+  return null;
```

Comment:
Tried something like:

```tsx
  const [container] = useState(() => {
    const el = document.createElement('div')
    return el
  })

  useEffect(() => {
    document.body.appendChild(container);
    return () => {
      document.body.removeChild(container)
    }
  }, [])

  return (
    <>{createPortal(<HeadlessMenu.Items {...optionProps} />, container)}</>
  );
```

but still got error along the lines of document is not defined

Reply:
Rather than finagling with this more I felt like checking for `window` was simpler and easier to understand